src/components/Chart/index.jsx

```diff
@@ -2,59 +2,44 @@
 
 import './styles.postcss';
 import React from 'react';
-import d3 from 'd3';
 import $ from 'jquery';
+import d3 from 'd3';
+import _ from 'underscore';
+import moment from 'moment';
+import monthDataCollection from './../../scripts/collection/MonthDataCollection';
 
 class Chart extends React.Component {
 
   constructor(props) {
     super(props);
+
     this.state = {
-      data: [
-          {
-            "day": 1,
-            "count": 0
-          },
-          {
-            "day": 3,
-            "count": 20
-          },
-          {
-            "day": 8,
-            "count": 55
-          },
-          {
-            "day": 14,
-            "count": 33
-          },
-          {
-            "day": 19,
-            "count": 72
-          },
-          {
-            "day": 24,
-            "count": 10
-          },
-          {
-            "day": 29,
-            "count": 99
-          }
-        ]
-    };
+      data: monthDataCollection.toJSON()
+    }
   }
 
   componentDidMount() {
-    this.setChart();
+    this._setChart();
+  }
+
+  componentDidUpdate() {
+    this._setChart();
   }
 
-  setChart() {
-    const data = this.state.data;
+  _getcurrentData() {
+    this.month = moment.utc(this.props.timelineDate).month();
```

Comment:
if you are not going to use this.month out of this function, it doesn't make any sense to have it in this way, change it for cons.